### PR TITLE
feat(GlobalHeader): Enterprise view

### PR DESCRIPTION
**Pull request by dreamwasp**
### Overview

Adds enterprise dropdown header variant to our GlobalHeader.

I also reorganized our GlobalHeader stories to be split by user type. The previous version took a very long time to load and was unwieldily. 

** Added disco as reviewers to check out the storybook + ensure their experiments are still working as intended

### PR Checklist

- [x] Related to designs: [here](https://www.figma.com/file/ReGfRNillGABAj5SlITalN/%F0%9F%93%90-Gamut?node-id=29718:38907&t=55uIyFk1ZfjZImEA-0) + [here](https://www.figma.com/file/Y2xXPfZRbJco3QW0dqsFLx/Module-(Course)-Syllabus-Page?node-id=1:3&t=wVwvlQyEqrxjLdXg-0)
- [x] Related to JIRA ticket: [web-2274]
- [x] I have run this code to verify it works
- [x] This PR includes unit tests for the code change
- [x] This PR includes testing instructions tests for the code change
- [x] The alpha package of this PR is passing end-to-end tests in all relevant Codecademy repositories

#### Testing Instructions

* Enterprise header should match the designs on mobile + desktop view.
* All header variants should be represented in the new stories. 
* Headers should look + act identically to previously in both the monolith + portal PR envs.

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](https://github.com/codecademy-engineering/Codecademy/pull/34917)  | [Monolith Env](https://pr-34917-monolith.dev-eks.codecademy.com/) |
| Portal       | [Portal Link](https://github.com/codecademy-engineering/mono/pull/1425)  | [Portal Env](https://tengu.codecademy.com/catalog?PR_ENV=portal-app-pr-1425)   |

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.290-alpha.042572.0<br/>@codecademy/gamut-labs@35.7.0-alpha.042572.0<br/>@codecademy/gamut@52.1.0-alpha.042572.0<br/>@codecademy/monaco@0.7.9-alpha.042572.0<br/>@codecademy/styleguide@63.6.0-alpha.042572.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://64077e0d3c9b2400b0d5589c--gamut-preview.netlify.app>https://64077e0d3c9b2400b0d5589c--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/64077e0d3c9b2400b0d5589c>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.290-alpha.281c70.0<br/>@codecademy/gamut-labs@35.7.0-alpha.281c70.0<br/>@codecademy/gamut@52.1.0-alpha.281c70.0<br/>@codecademy/monaco@0.7.9-alpha.281c70.0<br/>@codecademy/styleguide@63.6.0-alpha.281c70.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6407910825d779100b0c2c0f--gamut-preview.netlify.app>https://6407910825d779100b0c2c0f--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6407910825d779100b0c2c0f>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.290-alpha.14c854.0<br/>@codecademy/gamut-labs@35.7.0-alpha.14c854.0<br/>@codecademy/gamut@52.1.0-alpha.14c854.0<br/>@codecademy/monaco@0.7.9-alpha.14c854.0<br/>@codecademy/styleguide@63.6.0-alpha.14c854.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6407a05d085b7300bb38d6d6--gamut-preview.netlify.app>https://6407a05d085b7300bb38d6d6--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6407a05d085b7300bb38d6d6>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.291-alpha.0fc27b.0<br/>@codecademy/gamut-labs@35.8.0-alpha.0fc27b.0<br/>@codecademy/gamut@52.2.0-alpha.0fc27b.0<br/>@codecademy/monaco@0.7.10-alpha.0fc27b.0<br/>@codecademy/styleguide@63.7.0-alpha.0fc27b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6407a55c8f58c905e386d6e9--gamut-preview.netlify.app>https://6407a55c8f58c905e386d6e9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6407a55c8f58c905e386d6e9>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.291-alpha.eb21df.0<br/>@codecademy/gamut-labs@35.8.0-alpha.eb21df.0<br/>@codecademy/gamut@52.2.0-alpha.eb21df.0<br/>@codecademy/monaco@0.7.10-alpha.eb21df.0<br/>@codecademy/styleguide@63.7.0-alpha.eb21df.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6407abbcabb5240b2f9a24a2--gamut-preview.netlify.app>https://6407abbcabb5240b2f9a24a2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6407abbcabb5240b2f9a24a2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.291-alpha.dbf97c.0<br/>@codecademy/gamut-labs@35.8.0-alpha.dbf97c.0<br/>@codecademy/gamut@52.2.0-alpha.dbf97c.0<br/>@codecademy/monaco@0.7.10-alpha.dbf97c.0<br/>@codecademy/styleguide@63.7.0-alpha.dbf97c.0<br/></pre>

File: packages/gamut-labs/src/GlobalHeader/GlobalHeaderItems.tsx [hunk 1]
```diff
@@ -42,6 +53,17 @@ export const myHome: AppHeaderLinkItem = {
   type: 'link',
 };
 
+export const myPercipioHome: AppHeaderLinkItem = {
+  dataTestId: 'header-percihome',
+  icon: HouseEntranceIcon,
+  id: 'my-home',
+  text: 'My Percipio',
+  href: 'https://my.percipio.com/',
```
Comment (vonbarown) [inline packages/gamut-labs/src/GlobalHeader/GlobalHeaderItems.tsx:61]: Is this the link that we've settled and not the ``external_business_site`` ?

Comment (dreamwasp) [reply packages/gamut-labs/src/GlobalHeader/GlobalHeaderItems.tsx:61]: sorry i don't have a ton of knowledge here -- whats the external business site?

Comment (vonbarown) [reply packages/gamut-labs/src/GlobalHeader/GlobalHeaderItems.tsx:61]: It is the link that gets added to a enterprise users business plan that acts are a back link for account managers when the visit the BAM and click on the CTA to navigate back to percipio. For our percipio accounts it is ``https://my.percipio.com/``.  It was added to the ``user_business_module`` [here](https://github.com/codecademy-engineering/Codecademy/blob/243dd43fcf541ae76b66a7e492b476fac3f2c1ca/lib/user_business_module.rb#L86) and was already made available through the ``/web`` endpoint and the graphql query.

File: packages/gamut-labs/src/AppHeader/AppHeaderElements/AppHeaderLogo/index.tsx [hunk 2]
```diff
@@ -46,9 +46,15 @@ export const AppHeaderLogo: React.FC<AppHeaderLogoProps> = ({
       <Logo
         color="currentColor"
         variant={
-          item.checkMini && showMini ? 'mini' : item.pro ? 'pro' : 'default'
+          item.checkMini && showMini
```
Comment (saghdaey) [inline packages/gamut-labs/src/AppHeader/AppHeaderElements/AppHeaderLogo/index.tsx:49]: nit: maybe we can add a helper function to get the logo variant,  nested ternaries  are hard to read (imo!)

File: packages/styleguide/stories/Brand Labs/Organisms/GlobalHeader/Docs.stories.mdx [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+        <GlobalHeader {...args} />
+        <Box height="120vh" bg="background-primary" />
+      </Box>
+    )}
+  </Story>
+</Canvas>
+
+GlobalHeader determines whether to render the desktop or mobile version of the global header (<LinkTo kind="Molecules/AppHeader" story="AppHeader">AppHeader</LinkTo> or <LinkTo kind="Molecules/AppHeaderMobile" story="AppHeaderMobile">AppHeaderMobile</LinkTo>, respectively) based on the viewport size.
+
+GlobalHeader takes in several props. `type` determines which header to render based on the user role (anonymous, free, or pro), or a loading state. `type` will also determine which of Search and/or Notification components need to be passed in: Search will appear for all users, whereas Notifications will only appear for logged-in users.
```
[the remark this answers is not in the document]
Comment (dreamwasp) [reply packages/styleguide/stories/Brand Labs/Organisms/GlobalHeader/Docs.stories.mdx:60]: yes, good catch!

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.291-alpha.feb5b3.0<br/>@codecademy/gamut-labs@35.8.0-alpha.feb5b3.0<br/>@codecademy/gamut@52.2.0-alpha.feb5b3.0<br/>@codecademy/monaco@0.7.10-alpha.feb5b3.0<br/>@codecademy/styleguide@63.7.0-alpha.feb5b3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6408fec9ea20d0187a3121c2--gamut-preview.netlify.app>https://6408fec9ea20d0187a3121c2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6408fec9ea20d0187a3121c2>Deploy Logs</a></p>

**Review by saghdaey — APPROVED** [review]
LGTM! 

a complete aside when should we remove the "new" label from projects in the resources menu? IIRC projects were launched over a year ago :-O

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://64090bb3d23a651f3ab0983d--gamut-preview.netlify.app>https://64090bb3d23a651f3ab0983d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/64090bb3d23a651f3ab0983d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.291-alpha.44abef.0<br/>@codecademy/gamut-labs@35.8.0-alpha.44abef.0<br/>@codecademy/gamut@52.2.0-alpha.44abef.0<br/>@codecademy/monaco@0.7.10-alpha.44abef.0<br/>@codecademy/styleguide@63.7.0-alpha.44abef.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://64091021b8c2152433022feb--gamut-preview.netlify.app>https://64091021b8c2152433022feb--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/64091021b8c2152433022feb>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.291-alpha.158ecd.0<br/>@codecademy/gamut-labs@35.8.0-alpha.158ecd.0<br/>@codecademy/gamut@52.2.0-alpha.158ecd.0<br/>@codecademy/monaco@0.7.10-alpha.158ecd.0<br/>@codecademy/styleguide@63.7.0-alpha.158ecd.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://640b68f4e6e0e00f72989199--gamut-preview.netlify.app>https://640b68f4e6e0e00f72989199--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/640b68f4e6e0e00f72989199>Deploy Logs</a></p>

**Review by vonbarown — APPROVED** [review]
LGTM, thank you for adding this

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://640b760c27b70c1d163107b9--gamut-preview.netlify.app>https://640b760c27b70c1d163107b9--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/640b760c27b70c1d163107b9>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.291-alpha.36c179.0<br/>@codecademy/gamut-labs@35.8.0-alpha.36c179.0<br/>@codecademy/gamut@52.2.0-alpha.36c179.0<br/>@codecademy/monaco@0.7.10-alpha.36c179.0<br/>@codecademy/styleguide@63.7.0-alpha.36c179.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6410816da614940fae0530ec--gamut-preview.netlify.app>https://6410816da614940fae0530ec--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6410816da614940fae0530ec>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.291-alpha.8029db.0<br/>@codecademy/gamut-labs@35.8.0-alpha.8029db.0<br/>@codecademy/gamut@52.2.0-alpha.8029db.0<br/>@codecademy/monaco@0.7.10-alpha.8029db.0<br/>@codecademy/styleguide@63.7.0-alpha.8029db.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://6410981b64f2bb00be5e9028--gamut-preview.netlify.app>https://6410981b64f2bb00be5e9028--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/6410981b64f2bb00be5e9028>Deploy Logs</a></p>

**Pull request merged**